x-pack/plugins/siem/public/components/page/hosts/hosts_table/index.tsx

```diff
@@ -134,56 +155,92 @@ const getHostsColumns = (): Array<Columns<HostsEdges>> => [
               id,
               name: hostName,
               kqlQuery: '',
-              queryMatch: {
-                displayField: 'host.name',
-                displayValue: hostName,
-                field: 'host.id',
-                value: hostId,
-              },
-              queryDate: {
-                from: new Date(node.firstSeen!).valueOf(),
-                to: Date.now(),
-              },
+              queryMatch: { field: 'host.name', value: hostName },
+              queryDate: { from: startDate, to: Date.now() },
             }}
             render={(dataProvider, _, snapshot) =>
               snapshot.isDragging ? (
                 <DragEffects>
                   <Provider dataProvider={dataProvider} />
                 </DragEffects>
               ) : (
-                <HostDetailsLink hostId={hostId}>{hostName}</HostDetailsLink>
+                <AddToKql
+                  expression={`host.name: ${escapeQueryValue(hostName)}`}
+                  componentFilterType="hosts"
+                  type={type}
+                >
+                  <HostDetailsLink hostName={hostName}>{hostName}</HostDetailsLink>
+                </AddToKql>
               )
             }
           />
         );
-      } else {
-        return getEmptyTagValue();
       }
+      return getEmptyTagValue();
     },
   },
   {
     name: i18n.FIRST_SEEN,
     truncateText: false,
     hideForMobile: false,
-    render: ({ node }) =>
-      node.firstSeen && node.firstSeen !== '' ? (
-        <EuiToolTip position="bottom" content={node.firstSeen}>
-          <FormattedRelative value={node.firstSeen} />
-        </EuiToolTip>
-      ) : (
-        defaultToEmptyTag(node.firstSeen)
-      ),
+    render: ({ node }) => {
+      const hostname: string | null = get('host.name', node);
```

Comment:
👍 Nice! Gives the types on it. I was mentioning this above. This is very welcome.

Technically this should be `hostname: string | null undefined = get('host.name', node);` Since HostEcsFields says it can be undefined as well and lodash get can and will return undefined.

Reply:
Won't comment on the others below but the optional undefined is true for them as well.

Reply:
Fixed there and the ones below too